drivers/axi_core/clk_axi_clkgen/clk_axi_clkgen.c

```diff
@@ -368,12 +368,12 @@ void axi_clkgen_calc_params(struct axi_clkgen *axi_clkgen,
 			dout = DIV_ROUND_CLOSEST(fvco, fout);
 			dout = clamp(dout, 1, 128);
 			f = fvco / dout;
-			if (abs(f - fout) < abs(best_f - fout)) {
+			if (abs((int32_t)(f - fout)) < abs((int32_t)(best_f - fout))) {
```

Comment:
This looks a little bit strange to me... Since fout was already divided by 1000, maybe the safest thing would be to cast fout instead of the entire result.

Reply:
True. I edited the commit and casted `fout` only instead.
It seems safe because it is guaranteed to be less than 2^31-1 (max int32_t value) due to the division you mentioned.